crates/extensions/c8y_http_proxy/src/tests.rs

```diff
@@ -127,6 +133,58 @@ async fn retry_internal_id_on_expired_jwt() {
     .await;
 }
 
+#[tokio::test]
+async fn retry_internal_id_on_expired_jwt_with_mock() {
+    let device_id = "device-001";
+    let tmp_dir = "/tmp";
+    let external_id = "external-device-001";
+
+    let response = InternalIdResponse::new(device_id, external_id);
+    let response = serde_json::to_string(&response).unwrap();
+    // Start a lightweight mock server.
+    let mut server = mockito::Server::new();
+
+    let _mock1 = server
+        .mock("GET", "/identity/externalIds/c8y_Serial/device-001")
+        .match_header(
+            "Authorization",
+            Matcher::Exact("Bearer Cached JWT token".into()),
+        )
+        .with_status(401)
+        .create();
+    let _mock2 = server
+        .mock("GET", "/identity/externalIds/c8y_Serial/device-001")
+        .match_header(
+            "Authorization",
+            Matcher::Exact("Bearer Fresh JWT token".into()),
+        )
+        .with_status(200)
+        .with_body(response)
+        .create();
+
+    let mut target_url = server.url();
+    let mut jwt = ServerMessageBoxBuilder::new("JWT Actor", 16);
+
+    let mut http_actor = HttpActor::new().builder();
+
+    let config = C8YHttpConfig {
+        c8y_host: target_url.clone().into(),
+        device_id: device_id.into(),
+        tmp_dir: tmp_dir.into(),
+    };
+    let c8y_proxy_actor = C8YHttpProxyBuilder::new(config, &mut http_actor, &mut jwt);
+    let mut jwt_actor = ServerActor::new(DynamicJwtRetriever { count: 0 }, jwt.build());
+
+    tokio::spawn(async move { http_actor.run().await });
+    tokio::spawn(async move { jwt_actor.run().await });
+    let mut proxy = c8y_proxy_actor.build();
+
+    target_url.push_str("/identity/externalIds/c8y_Serial/device-001");
+    let result = proxy.try_get_internal_id(device_id.into()).await;
+    assert_eq!(device_id, result.unwrap());
```

Comment:
Isn't `try_get_internal_id` supposed to return an external_device_id and not a device_id ?

I was expecting the following:

```suggestion
    let result = proxy.try_get_internal_id(device_id.into()).await;
    assert_eq!(external_id, result.unwrap());
```

This test being green, it raises more concerns than it gives confidence.

Reply:
If this is is related to the events REST api then it is the Cumulocity global id (e.g. internal id, not the external identity; example 12345). The REST api users the global id exclusively (except for  the external identity loop up api).

Reply:
ah now I see the problem. The assertion is comparing the `device_id` with the result of the `try_get_internal_id` which should not be equal.

Reply:
I wrongly populated the struct `InternalIdResponse::new(internal_id,external_id)`.